### PR TITLE
Fail2ban - add daily ban notifications

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
This was successfully tested, hence marking as ready for review

File: apps/fail2ban.sh [hunk 1]
```diff
@@ -184,4 +186,46 @@ msg_box "Fail2ban is now successfully installed.
...
+crontab -u root -l | grep -v "$SCRIPTS/daily_fail2ban_report.sh"  | crontab -u root -
+crontab -u root -l | { cat; echo "59 23 * * * $SCRIPTS/daily_fail2ban_report.sh > /dev/null"; } | crontab -u root -
+
+# Adjust access rights
+chown root:root "$SCRIPTS/daily_fail2ban_report.sh"
+chmod 700 "$SCRIPTS/daily_fail2ban_report.sh"
+
+# Inform user
+msg_box "The daily Fail2Ban report was successfully configured.\n
+You will get notified at 23.59h, if new bans were made that day."
```
Comment (enoch85) [inline apps/fail2ban.sh:229]: use : not . please.

23:59

Comment (szaimen) [reply apps/fail2ban.sh]: Done

File: apps/fail2ban.sh [hunk 2]
```diff
@@ -184,4 +186,46 @@ msg_box "Fail2ban is now successfully installed.
 Please use 'fail2ban-client set nextcloud unbanip <Banned IP>' to unban certain IPs
 You can also use 'iptables -L -n' to check which IPs that are banned"
 
+# Daily ban notification
+if ! yesno_box_no "Do you want to get notified about daily bans?\n
+If you choose 'yes', you will receive a notification about daily bans at 23.59h."
```
Comment (enoch85) [inline apps/fail2ban.sh:191]: Same here ;)

Comment (szaimen) [reply apps/fail2ban.sh]: Done

**Comment by enoch85** [comment]
cc @tyleroconnell  for grammar and spelling.

Thanks! :+1:

**Comment by tyleroconnell** [comment]
Sorry @enoch85 I did not realize you were charging for downloads, I was just trying to contribute back to the open source community in a small way. Best of luck!

**Comment by enoch85** [comment]
@tyleroconnell :open_mouth: It's free and have always been: https://www.hanssonit.se/nextcloud-vm/ (40 GB)

File: apps/fail2ban.sh [hunk 3]
```diff
@@ -184,4 +186,46 @@ msg_box "Fail2ban is now successfully installed.
...
+if [ -z "\$BANNED_IPS" ]
+then
+    exit
+fi
+
+# Report if something was found
+source /var/scripts/fetch_lib.sh || source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+notify_admin_gui \
+"Your daily fail2ban report" \
+"Those got banned today:
```
Comment (enoch85) [inline apps/fail2ban.sh:215]: Please change to: These IP's got banned today:

Comment (szaimen) [reply apps/fail2ban.sh]: Done

**Pull request merged**